### ISSUE TITLE
"Inconsitent Pipeline" error when workspace is open

### ISSUE THREAD

**Issue by Cynical-Optimist** [buildstream-migration/bst-staging#576]
[See original issue on GitLab](https://gitlab.com/BuildStream/buildstream/-/issues/576)
In GitLab by [[Gitlab user @adds68]](https://gitlab.com/adds68) on Aug 14, 2018, 21:35

## Summary

When trying to build an element that has had a workspace open against it, i am seeing this error:

```
Inconsistent pipeline

    Exact versions are missing for the following elements
    Try tracking these elements first with `bst track`

      desktop/sdl2.bst

```

## Steps to reproduce
After some investigation i found these steps caused the issues:

* modify a .bst element
* create a workspace 
* build the element
* rm -r the workspace directory
* try to build the element again

## What is the current bug behavior?

When you try to build the element after removing the workspace directory you see the error above, and running bst track on the element does not fox the dependency issue.

When you close the bst workspace, the error no longer occurs and the element builds as expected.

## What is the expected correct behavior?

Alert the user that a workspace is open against this element and that they should close it.

## Relevant logs and/or screenshots

Bst error

```
Inconsistent pipeline

    Exact versions are missing for the following elements
    Try tracking these elements first with `bst track`

      desktop/sdl2.bst

```

## Possible fixes

[//]: # (If you can, link to the line of code that might be responsible for the problem)

## Other relevant information

[//]: # (Add the BuildStream version. Substitute below the "x" by "1", "2" or the right version. Add more than one version if necessary)

* BuildStream version affected: /milestone %"BuildStream\_v1.2" x

----

**Comment by Cynical-Optimist** [comment]
In GitLab by [[Gitlab user @adds68]](https://gitlab.com/adds68) on Aug 14, 2018, 21:36

changed the description

**Comment by Cynical-Optimist** [comment]
In GitLab by [[Gitlab user @adds68]](https://gitlab.com/adds68) on Aug 20, 2018, 09:28

changed the description

**Comment by Cynical-Optimist** [comment]
In GitLab by [[Gitlab user @toscalix]](https://gitlab.com/toscalix) on Aug 20, 2018, 09:31

assigned to [[Gitlab user @valentindavid]](https://gitlab.com/valentindavid)

**Comment by Cynical-Optimist** [comment]
In GitLab by [[Gitlab user @tristanvb]](https://gitlab.com/tristanvb) on Aug 20, 2018, 09:34

Drive by comment during our meeting discussing this:

* The fact that an element is considered *inconsistent* when it has an open but empty workspace, is expected
* The error message here is definitely confusing, and should be improved for this.

Maybe the best option is to fire an error for any empty workspaces, which would be raised before hitting this error.

**Comment by Cynical-Optimist** [comment]
In GitLab by [[Gitlab user @valentindavid]](https://gitlab.com/valentindavid) on Aug 22, 2018, 19:32

mentioned in commit 1479b32c75e8d1153be148ce3b06b26e743f8da8

**Comment by Cynical-Optimist** [comment]
In GitLab by [[Gitlab user @valentindavid]](https://gitlab.com/valentindavid) on Aug 22, 2018, 19:34

mentioned in merge request !703

**Comment by Cynical-Optimist** [comment]
In GitLab by [[Gitlab user @valentindavid]](https://gitlab.com/valentindavid) on Aug 23, 2018, 07:40

mentioned in commit 2215859c925c48dfebd1315b4f5ad2834689ea22

**Comment by Cynical-Optimist** [comment]
In GitLab by [[Gitlab user @valentindavid]](https://gitlab.com/valentindavid) on Aug 23, 2018, 08:06

mentioned in commit 2a33f557b6c3c940bdef3b539d7117c439ecc1ac

**Comment by Cynical-Optimist** [comment]
In GitLab by [[Gitlab user @tristanvb]](https://gitlab.com/tristanvb) on Aug 23, 2018, 08:45

mentioned in commit e37ac3bcfa00b29cadc5509898fd6946b8e76132

**Comment by Cynical-Optimist** [comment]
In GitLab by [[Gitlab user @valentindavid]](https://gitlab.com/valentindavid) on Aug 23, 2018, 08:45

closed via commit 2215859c925c48dfebd1315b4f5ad2834689ea22

**Comment by Cynical-Optimist** [comment]
In GitLab by [[Gitlab user @tristanvb]](https://gitlab.com/tristanvb) on Aug 23, 2018, 08:45

closed via merge request !703
